In [1]:
import pandas as pd
import numpy as np

## Ekstraksi Data bertipe Tanggal - DateTime

In [2]:
df = pd.read_csv('data.csv')
df.head()

,receipt_id,receipts_item_id,purchase_time,category,sub_category,format,unit_price,discount,quantity
0,9622257,32369294,7/22/2018 21:19,Rice,Rice,supermarket,128000.0,0,1
1,9446359,31885876,7/15/2018 16:17,Rice,Rice,minimarket,102750.0,0,1
2,9470290,31930241,7/15/2018 12:12,Rice,Rice,supermarket,64000.0,0,3
3,9643416,32418582,7/24/2018 8:27,Rice,Rice,minimarket,65000.0,0,1
4,9692093,32561236,7/26/2018 11:28,Rice,Rice,supermarket,124500.0,0,1


In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 72000 entries, 0 to 71999
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   receipt_id        72000 non-null  int64  
 1   receipts_item_id  72000 non-null  int64  
 2   purchase_time     72000 non-null  object 
 3   category          72000 non-null  object 
 4   sub_category      72000 non-null  object 
 5   format            72000 non-null  object 
 6   unit_price        72000 non-null  float64
 7   discount          72000 non-null  int64  
 8   quantity          72000 non-null  int64  
dtypes: float64(1), int64(4), object(4)
memory usage: 4.9+ MB


## Kolom tanggal, Hanya dapat di ekstraksi jika bertipe data **Datetime**
- Secara default Pandas akan membaca Kolom selain Numerik sebagai Object, Sehingga perlu kita konversi

In [4]:
## Konversi Data

## Alt 1
- Menggunakan parameter **parse_dates = [Nama Kolom]** Ketika awal membaca file

In [5]:
df = pd.read_csv('data.csv', index_col = 0, parse_dates=['purchase_time'])
df.head()

,receipts_item_id,purchase_time,category,sub_category,format,unit_price,discount,quantity
receipt_id,,,,,,,,
9622257,32369294,2018-07-22 21:19:00,Rice,Rice,supermarket,128000.0,0,1
9446359,31885876,2018-07-15 16:17:00,Rice,Rice,minimarket,102750.0,0,1
9470290,31930241,2018-07-15 12:12:00,Rice,Rice,supermarket,64000.0,0,3
9643416,32418582,2018-07-24 08:27:00,Rice,Rice,minimarket,65000.0,0,1
9692093,32561236,2018-07-26 11:28:00,Rice,Rice,supermarket,124500.0,0,1


In [6]:
df.dtypes

receipts_item_id             int64
purchase_time       datetime64[ns]
category                    object
sub_category                object
format                      object
unit_price                 float64
discount                     int64
quantity                     int64
dtype: object

## Alt 2
- menggu7nakan fungsi .astype()
- astype digunakan untuk mengubah tipe data, tidak terbatas pada date time
- .astype(datetime)

In [7]:
df = pd.read_csv('data.csv', index_col =0)
df.head()

,receipts_item_id,purchase_time,category,sub_category,format,unit_price,discount,quantity
receipt_id,,,,,,,,
9622257,32369294,7/22/2018 21:19,Rice,Rice,supermarket,128000.0,0,1
9446359,31885876,7/15/2018 16:17,Rice,Rice,minimarket,102750.0,0,1
9470290,31930241,7/15/2018 12:12,Rice,Rice,supermarket,64000.0,0,3
9643416,32418582,7/24/2018 8:27,Rice,Rice,minimarket,65000.0,0,1
9692093,32561236,7/26/2018 11:28,Rice,Rice,supermarket,124500.0,0,1


In [8]:
df.dtypes

receipts_item_id      int64
purchase_time        object
category             object
sub_category         object
format               object
unit_price          float64
discount              int64
quantity              int64
dtype: object

In [11]:
df['purchase_time'].astype('datetime64[ns]')

receipt_id
9622257   2018-07-22 21:19:00
9446359   2018-07-15 16:17:00
9470290   2018-07-15 12:12:00
9643416   2018-07-24 08:27:00
9692093   2018-07-26 11:28:00
                  ...        
5909305   2017-12-27 09:20:00
5736299   2017-12-13 19:52:00
5901144   2017-12-27 08:03:00
5660630   2017-12-07 12:29:00
5808147   2017-12-19 18:59:00
Name: purchase_time, Length: 72000, dtype: datetime64[ns]

In [12]:
df['purchase_time'] = df['purchase_time'].astype('datetime64[ns]')

In [13]:
df.dtypes

receipts_item_id             int64
purchase_time       datetime64[ns]
category                    object
sub_category                object
format                      object
unit_price                 float64
discount                     int64
quantity                     int64
dtype: object

## Alt 3
- Menggunakan fungsi **pd.to_datetime(nama kolom)**

In [14]:
df = pd.read_csv('data.csv', index_col =0)
df.head()

,receipts_item_id,purchase_time,category,sub_category,format,unit_price,discount,quantity
receipt_id,,,,,,,,
9622257,32369294,7/22/2018 21:19,Rice,Rice,supermarket,128000.0,0,1
9446359,31885876,7/15/2018 16:17,Rice,Rice,minimarket,102750.0,0,1
9470290,31930241,7/15/2018 12:12,Rice,Rice,supermarket,64000.0,0,3
9643416,32418582,7/24/2018 8:27,Rice,Rice,minimarket,65000.0,0,1
9692093,32561236,7/26/2018 11:28,Rice,Rice,supermarket,124500.0,0,1


In [15]:
df.dtypes

receipts_item_id      int64
purchase_time        object
category             object
sub_category         object
format               object
unit_price          float64
discount              int64
quantity              int64
dtype: object

In [16]:
pd.to_datetime(df['purchase_time'])

receipt_id
9622257   2018-07-22 21:19:00
9446359   2018-07-15 16:17:00
9470290   2018-07-15 12:12:00
9643416   2018-07-24 08:27:00
9692093   2018-07-26 11:28:00
                  ...        
5909305   2017-12-27 09:20:00
5736299   2017-12-13 19:52:00
5901144   2017-12-27 08:03:00
5660630   2017-12-07 12:29:00
5808147   2017-12-19 18:59:00
Name: purchase_time, Length: 72000, dtype: datetime64[ns]

In [17]:
df['purchase_time'] = pd.to_datetime(df['purchase_time'])

In [18]:
df.dtypes

receipts_item_id             int64
purchase_time       datetime64[ns]
category                    object
sub_category                object
format                      object
unit_price                 float64
discount                     int64
quantity                     int64
dtype: object

In [19]:
tanggal = pd.Series([
'31-03-2022',
'01-04-2022',
'02-04-2022',
'17-08-2022',
'31-10-2022',
'03-04-2022'])

In [20]:
tanggal

0    31-03-2022
1    01-04-2022
2    02-04-2022
3    17-08-2022
4    31-10-2022
5    03-04-2022
dtype: object

In [22]:
import warnings
warnings.filterwarnings('ignore')

In [23]:
tanggal.astype('datetime64[ns]')

0   2022-03-31
1   2022-04-01
2   2022-04-02
3   2022-08-17
4   2022-10-31
5   2022-04-03
dtype: datetime64[ns]

In [24]:
pd.to_datetime(tanggal)

0   2022-03-31
1   2022-04-01
2   2022-04-02
3   2022-08-17
4   2022-10-31
5   2022-04-03
dtype: datetime64[ns]

In [25]:
tgl = pd.Series([
'01-maret-2005',
'july/17/2017',
'Agustus-20-2021',
'20-september-2003',
'19-August-2016',
'20/May/2004',
'2021-02-15',
'07/25/2000',
'24-12-22',
'januari-01-2023'])
tgl

0        01-maret-2005
1         july/17/2017
2      Agustus-20-2021
3    20-september-2003
4       19-August-2016
5          20/May/2004
6           2021-02-15
7           07/25/2000
8             24-12-22
9      januari-01-2023
dtype: object

In [27]:
#tgl.astype('datetime64[ns]')
## Error

In [32]:
#pd.to_datetime(tgl, yearfirst = True)
#pd.to_datetime(tgl, dayfirst=True)

In [35]:
import dateparser

In [34]:
!pip install dateparser

Defaulting to user installation because normal site-packages is not writeable
  Obtaining dependency information for dateparser from https://files.pythonhosted.org/packages/a4/29/db12aa4dda81580be1999824a689bd52aa40061fc12c9ccdc3feab5ea718/dateparser-1.2.0-py2.py3-none-any.whl.metadata
  Obtaining dependency information for tzlocal from https://files.pythonhosted.org/packages/97/3f/c4c51c55ff8487f2e6d0e618dba917e3c3ee2caae6cf0fbb59c9b1876f2e/tzlocal-5.2-py3-none-any.whl.metadata
Using cached dateparser-1.2.0-py2.py3-none-any.whl (294 kB)
Using cached tzlocal-5.2-py3-none-any.whl (17 kB)


  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.


In [36]:
tgl

0        01-maret-2005
1         july/17/2017
2      Agustus-20-2021
3    20-september-2003
4       19-August-2016
5          20/May/2004
6           2021-02-15
7           07/25/2000
8             24-12-22
9      januari-01-2023
dtype: object

In [37]:
tgl.apply(dateparser.parse)

0   2005-03-01
1   2017-07-17
2   2021-08-20
3   2003-09-20
4   2016-08-19
5   2004-05-20
6   2021-02-15
7   2000-07-25
8   2022-12-24
9   2023-01-01
dtype: datetime64[ns]

## Proses Ekstraksi

In [38]:
df.head()

,receipts_item_id,purchase_time,category,sub_category,format,unit_price,discount,quantity
receipt_id,,,,,,,,
9622257,32369294,2018-07-22 21:19:00,Rice,Rice,supermarket,128000.0,0,1
9446359,31885876,2018-07-15 16:17:00,Rice,Rice,minimarket,102750.0,0,1
9470290,31930241,2018-07-15 12:12:00,Rice,Rice,supermarket,64000.0,0,3
9643416,32418582,2018-07-24 08:27:00,Rice,Rice,minimarket,65000.0,0,1
9692093,32561236,2018-07-26 11:28:00,Rice,Rice,supermarket,124500.0,0,1


In [39]:
df.dtypes

receipts_item_id             int64
purchase_time       datetime64[ns]
category                    object
sub_category                object
format                      object
unit_price                 float64
discount                     int64
quantity                     int64
dtype: object

In [40]:
purchase_dt = df['purchase_time'].dt

In [41]:
### Ambil - Ekstrak Data Bulan berupa Angka
purchase_dt.month

receipt_id
9622257     7
9446359     7
9470290     7
9643416     7
9692093     7
           ..
5909305    12
5736299    12
5901144    12
5660630    12
5808147    12
Name: purchase_time, Length: 72000, dtype: int32

In [42]:
## Memasukkan data hasil ekstraksi ke dalam dataframe

In [43]:
df['month'] = purchase_dt.month

In [44]:
df.head()

,receipts_item_id,purchase_time,category,sub_category,format,unit_price,discount,quantity,month
receipt_id,,,,,,,,,
9622257,32369294,2018-07-22 21:19:00,Rice,Rice,supermarket,128000.0,0,1,7
9446359,31885876,2018-07-15 16:17:00,Rice,Rice,minimarket,102750.0,0,1,7
9470290,31930241,2018-07-15 12:12:00,Rice,Rice,supermarket,64000.0,0,3,7
9643416,32418582,2018-07-24 08:27:00,Rice,Rice,minimarket,65000.0,0,1,7
9692093,32561236,2018-07-26 11:28:00,Rice,Rice,supermarket,124500.0,0,1,7


In [45]:
## Ambil data bulan berupa nama
purchase_dt.month_name()

receipt_id
9622257        July
9446359        July
9470290        July
9643416        July
9692093        July
             ...   
5909305    December
5736299    December
5901144    December
5660630    December
5808147    December
Name: purchase_time, Length: 72000, dtype: object

In [46]:
## Memasukkan hasil ekstraksi ke dataframe
df['bulan'] = purchase_dt.month_name()

In [47]:
df.head()

,receipts_item_id,purchase_time,category,sub_category,format,unit_price,discount,quantity,month,bulan
receipt_id,,,,,,,,,,
9622257,32369294,2018-07-22 21:19:00,Rice,Rice,supermarket,128000.0,0,1,7,July
9446359,31885876,2018-07-15 16:17:00,Rice,Rice,minimarket,102750.0,0,1,7,July
9470290,31930241,2018-07-15 12:12:00,Rice,Rice,supermarket,64000.0,0,3,7,July
9643416,32418582,2018-07-24 08:27:00,Rice,Rice,minimarket,65000.0,0,1,7,July
9692093,32561236,2018-07-26 11:28:00,Rice,Rice,supermarket,124500.0,0,1,7,July


In [48]:
## ambil data Tahun
purchase_dt.year

receipt_id
9622257    2018
9446359    2018
9470290    2018
9643416    2018
9692093    2018
           ... 
5909305    2017
5736299    2017
5901144    2017
5660630    2017
5808147    2017
Name: purchase_time, Length: 72000, dtype: int32

In [49]:
df['year'] = purchase_dt.year

In [50]:
## Ambil data Tanggal
purchase_dt.day

receipt_id
9622257    22
9446359    15
9470290    15
9643416    24
9692093    26
           ..
5909305    27
5736299    13
5901144    27
5660630     7
5808147    19
Name: purchase_time, Length: 72000, dtype: int32

In [51]:
## Ambil Hari - berupa angka
purchase_dt.dayofweek
## Berurutan senin - minggu, senin = 0, minggu = 6

receipt_id
9622257    6
9446359    6
9470290    6
9643416    1
9692093    3
          ..
5909305    2
5736299    2
5901144    2
5660630    3
5808147    1
Name: purchase_time, Length: 72000, dtype: int32

In [52]:
## Ambil Hari - berupa Nama
purchase_dt.day_name()

receipt_id
9622257       Sunday
9446359       Sunday
9470290       Sunday
9643416      Tuesday
9692093     Thursday
             ...    
5909305    Wednesday
5736299    Wednesday
5901144    Wednesday
5660630     Thursday
5808147      Tuesday
Name: purchase_time, Length: 72000, dtype: object

In [53]:
## memasukkan data ke Dataframe
df['day'] = purchase_dt.day_name()

In [54]:
df['hari'] = purchase_dt.dayofweek

In [55]:
df.head()

,receipts_item_id,purchase_time,category,sub_category,format,unit_price,discount,quantity,month,bulan,year,day,hari
receipt_id,,,,,,,,,,,,,
9622257,32369294,2018-07-22 21:19:00,Rice,Rice,supermarket,128000.0,0,1,7,July,2018,Sunday,6
9446359,31885876,2018-07-15 16:17:00,Rice,Rice,minimarket,102750.0,0,1,7,July,2018,Sunday,6
9470290,31930241,2018-07-15 12:12:00,Rice,Rice,supermarket,64000.0,0,3,7,July,2018,Sunday,6
9643416,32418582,2018-07-24 08:27:00,Rice,Rice,minimarket,65000.0,0,1,7,July,2018,Tuesday,1
9692093,32561236,2018-07-26 11:28:00,Rice,Rice,supermarket,124500.0,0,1,7,July,2018,Thursday,3


In [56]:
def WeekEnd(x):
    if x > 4:
        return "Yes"
    else:
        return "No"

In [58]:
df['hari'].apply(WeekEnd)

receipt_id
9622257    Yes
9446359    Yes
9470290    Yes
9643416     No
9692093     No
          ... 
5909305     No
5736299     No
5901144     No
5660630     No
5808147     No
Name: hari, Length: 72000, dtype: object

In [59]:
df['WeekEnd'] = df['hari'].apply(WeekEnd)

In [60]:
df.head()

,receipts_item_id,purchase_time,category,sub_category,format,unit_price,discount,quantity,month,bulan,year,day,hari,WeekEnd
receipt_id,,,,,,,,,,,,,,
9622257,32369294,2018-07-22 21:19:00,Rice,Rice,supermarket,128000.0,0,1,7,July,2018,Sunday,6,Yes
9446359,31885876,2018-07-15 16:17:00,Rice,Rice,minimarket,102750.0,0,1,7,July,2018,Sunday,6,Yes
9470290,31930241,2018-07-15 12:12:00,Rice,Rice,supermarket,64000.0,0,3,7,July,2018,Sunday,6,Yes
9643416,32418582,2018-07-24 08:27:00,Rice,Rice,minimarket,65000.0,0,1,7,July,2018,Tuesday,1,No
9692093,32561236,2018-07-26 11:28:00,Rice,Rice,supermarket,124500.0,0,1,7,July,2018,Thursday,3,No


In [61]:
## Ambil Data Jam
purchase_dt.hour

receipt_id
9622257    21
9446359    16
9470290    12
9643416     8
9692093    11
           ..
5909305     9
5736299    19
5901144     8
5660630    12
5808147    18
Name: purchase_time, Length: 72000, dtype: int32

In [62]:
## ambil hanya data tanggal (date)
purchase_dt.to_period('D')

receipt_id
9622257    2018-07-22
9446359    2018-07-15
9470290    2018-07-15
9643416    2018-07-24
9692093    2018-07-26
              ...    
5909305    2017-12-27
5736299    2017-12-13
5901144    2017-12-27
5660630    2017-12-07
5808147    2017-12-19
Name: purchase_time, Length: 72000, dtype: period[D]

In [63]:
## Ambil data Bulan - Tahun
purchase_dt.to_period('M')

receipt_id
9622257    2018-07
9446359    2018-07
9470290    2018-07
9643416    2018-07
9692093    2018-07
            ...   
5909305    2017-12
5736299    2017-12
5901144    2017-12
5660630    2017-12
5808147    2017-12
Name: purchase_time, Length: 72000, dtype: period[M]

In [64]:
## Ambil data Weekly - Periode Mingguan (senin - minggu)

In [66]:
purchase_dt.to_period('W')

receipt_id
9622257    2018-07-16/2018-07-22
9446359    2018-07-09/2018-07-15
9470290    2018-07-09/2018-07-15
9643416    2018-07-23/2018-07-29
9692093    2018-07-23/2018-07-29
                   ...          
5909305    2017-12-25/2017-12-31
5736299    2017-12-11/2017-12-17
5901144    2017-12-25/2017-12-31
5660630    2017-12-04/2017-12-10
5808147    2017-12-18/2017-12-24
Name: purchase_time, Length: 72000, dtype: period[W-SUN]

In [67]:
## Ambil data Kuartal
purchase_dt.to_period('Q')

receipt_id
9622257    2018Q3
9446359    2018Q3
9470290    2018Q3
9643416    2018Q3
9692093    2018Q3
            ...  
5909305    2017Q4
5736299    2017Q4
5901144    2017Q4
5660630    2017Q4
5808147    2017Q4
Name: purchase_time, Length: 72000, dtype: period[Q-DEC]

In [68]:
df['quarter'] = purchase_dt.to_period('Q')

In [69]:
df['yearMonth'] = purchase_dt.to_period('M')

In [70]:
df.head()

,receipts_item_id,purchase_time,category,sub_category,format,unit_price,discount,quantity,month,bulan,year,day,hari,WeekEnd,quarter,yearMonth
receipt_id,,,,,,,,,,,,,,,,
9622257,32369294,2018-07-22 21:19:00,Rice,Rice,supermarket,128000.0,0,1,7,July,2018,Sunday,6,Yes,2018Q3,2018-07
9446359,31885876,2018-07-15 16:17:00,Rice,Rice,minimarket,102750.0,0,1,7,July,2018,Sunday,6,Yes,2018Q3,2018-07
9470290,31930241,2018-07-15 12:12:00,Rice,Rice,supermarket,64000.0,0,3,7,July,2018,Sunday,6,Yes,2018Q3,2018-07
9643416,32418582,2018-07-24 08:27:00,Rice,Rice,minimarket,65000.0,0,1,7,July,2018,Tuesday,1,No,2018Q3,2018-07
9692093,32561236,2018-07-26 11:28:00,Rice,Rice,supermarket,124500.0,0,1,7,July,2018,Thursday,3,No,2018Q3,2018-07


In [71]:
df['tgl'] = purchase_dt.day

In [72]:
df['hour'] = purchase_dt.hour

In [73]:
df.head()

,receipts_item_id,purchase_time,category,sub_category,format,unit_price,discount,quantity,month,bulan,year,day,hari,WeekEnd,quarter,yearMonth,tgl,hour
receipt_id,,,,,,,,,,,,,,,,,,
9622257,32369294,2018-07-22 21:19:00,Rice,Rice,supermarket,128000.0,0,1,7,July,2018,Sunday,6,Yes,2018Q3,2018-07,22,21
9446359,31885876,2018-07-15 16:17:00,Rice,Rice,minimarket,102750.0,0,1,7,July,2018,Sunday,6,Yes,2018Q3,2018-07,15,16
9470290,31930241,2018-07-15 12:12:00,Rice,Rice,supermarket,64000.0,0,3,7,July,2018,Sunday,6,Yes,2018Q3,2018-07,15,12
9643416,32418582,2018-07-24 08:27:00,Rice,Rice,minimarket,65000.0,0,1,7,July,2018,Tuesday,1,No,2018Q3,2018-07,24,8
9692093,32561236,2018-07-26 11:28:00,Rice,Rice,supermarket,124500.0,0,1,7,July,2018,Thursday,3,No,2018Q3,2018-07,26,11


In [74]:
df['total_payment'] = (df['unit_price'] * df['quantity']) - df['discount']

In [77]:
purchase_dt.strftime('%B / %H:%M')
## Custom format sesuai kebutuhan
## Format mengacu pada format Date time pada python ==> strftime.org

receipt_id
9622257        July / 21:19
9446359        July / 16:17
9470290        July / 12:12
9643416        July / 08:27
9692093        July / 11:28
                 ...       
5909305    December / 09:20
5736299    December / 19:52
5901144    December / 08:03
5660630    December / 12:29
5808147    December / 18:59
Name: purchase_time, Length: 72000, dtype: object

In [78]:
df['custom'] = purchase_dt.strftime('%B / %H:%M')

In [79]:
df.head()

,receipts_item_id,purchase_time,category,sub_category,format,unit_price,discount,quantity,month,bulan,year,day,hari,WeekEnd,quarter,yearMonth,tgl,hour,total_payment,custom
receipt_id,,,,,,,,,,,,,,,,,,,,
9622257,32369294,2018-07-22 21:19:00,Rice,Rice,supermarket,128000.0,0,1,7,July,2018,Sunday,6,Yes,2018Q3,2018-07,22,21,128000.0,July / 21:19
9446359,31885876,2018-07-15 16:17:00,Rice,Rice,minimarket,102750.0,0,1,7,July,2018,Sunday,6,Yes,2018Q3,2018-07,15,16,102750.0,July / 16:17
9470290,31930241,2018-07-15 12:12:00,Rice,Rice,supermarket,64000.0,0,3,7,July,2018,Sunday,6,Yes,2018Q3,2018-07,15,12,192000.0,July / 12:12
9643416,32418582,2018-07-24 08:27:00,Rice,Rice,minimarket,65000.0,0,1,7,July,2018,Tuesday,1,No,2018Q3,2018-07,24,8,65000.0,July / 08:27
9692093,32561236,2018-07-26 11:28:00,Rice,Rice,supermarket,124500.0,0,1,7,July,2018,Thursday,3,No,2018Q3,2018-07,26,11,124500.0,July / 11:28


In [80]:
## Analisis Data

- Crosstab
- Group by
- Pivot Table

In [81]:
## Frequency Table

In [82]:
## Univariate Analysis -= Analisis per Kolom

In [83]:
df.head()

,receipts_item_id,purchase_time,category,sub_category,format,unit_price,discount,quantity,month,bulan,year,day,hari,WeekEnd,quarter,yearMonth,tgl,hour,total_payment,custom
receipt_id,,,,,,,,,,,,,,,,,,,,
9622257,32369294,2018-07-22 21:19:00,Rice,Rice,supermarket,128000.0,0,1,7,July,2018,Sunday,6,Yes,2018Q3,2018-07,22,21,128000.0,July / 21:19
9446359,31885876,2018-07-15 16:17:00,Rice,Rice,minimarket,102750.0,0,1,7,July,2018,Sunday,6,Yes,2018Q3,2018-07,15,16,102750.0,July / 16:17
9470290,31930241,2018-07-15 12:12:00,Rice,Rice,supermarket,64000.0,0,3,7,July,2018,Sunday,6,Yes,2018Q3,2018-07,15,12,192000.0,July / 12:12
9643416,32418582,2018-07-24 08:27:00,Rice,Rice,minimarket,65000.0,0,1,7,July,2018,Tuesday,1,No,2018Q3,2018-07,24,8,65000.0,July / 08:27
9692093,32561236,2018-07-26 11:28:00,Rice,Rice,supermarket,124500.0,0,1,7,July,2018,Thursday,3,No,2018Q3,2018-07,26,11,124500.0,July / 11:28


## 1. Business Question : Sub Category apa yg memiliki Penjualan (items) Tertinggi?

In [84]:
df['sub_category'].value_counts()

sub_category
Detergent    36000
Sugar        24000
Rice         12000
Name: count, dtype: int64

In [85]:
pd.crosstab(df['sub_category'], columns='Jumlah')

col_0,Jumlah
sub_category,
Detergent,36000
Rice,12000
Sugar,24000


- **Insight** - Informasi : Penjualan tertinggi (items) dari Sub category detergent sebanyak 36.000 transaksi

## 2. Business Question : Sub Category apa yg memiliki penjualan (items) paling sedikit?

In [86]:
df['sub_category'].value_counts()

sub_category
Detergent    36000
Sugar        24000
Rice         12000
Name: count, dtype: int64

In [89]:
pd.crosstab(df['sub_category'], columns='Jumlah').sort_values('Jumlah')

col_0,Jumlah
sub_category,
Rice,12000
Sugar,24000
Detergent,36000


- **Insight** - Informasi : Penjualan paling sedikit (items) dari sub category Rice sebanyak 12.000 transaksi

## 3. Business Question : Lokasi (format) mana yg memiliki penjualan paling sedikit?

In [90]:
df['format'].value_counts()

format
minimarket     46803
supermarket    19826
hypermarket     5371
Name: count, dtype: int64

- Insight - Informasi :
    - Penjualan paling sedikit (items) dari lokasi/format Hypermarket sebanyak 5371
    - Penjualan paling tinggi ada di Minimarket sebanyak 46.803, karena lokasi minimarket dekat dg perumahan dan jumlahnya cukup banyak. rata-rata jarakl 100m dari perumahan, jumlah minimal ada 2 minimarket
    
source : https://bisnis.co.id/rata-rata-jarak-minimarket

In [91]:
pd.crosstab(df['format'], columns='Jumlah').sort_values('Jumlah')

col_0,Jumlah
format,
hypermarket,5371
supermarket,19826
minimarket,46803


## Insight Data - Informasi :
- Transaksi paling tinggi ada di minimarket, sedangkan paling rendah ada di hypermarket
- Jika akan merilis produk baru, sebaikonya dilakukan di minimarket
- Jika akan melakukan promo produk sebaiknya di minimarket
- Penjualan di minimarket tinggi karena lokasi minimarket dekat dg perumahan dan jumlahnya cukup banyak. rata-rata jarakl 100m dari perumahan, jumlah minimal ada 2 minimarket ==> Hipotesa/asumsi
- Jika ada Hipotesa - dugaan - asumsi harus berdasarkan data

Source : .....